app/workers/travis_build_killer.rb

```diff
@@ -0,0 +1,35 @@
+class TravisBuildKiller
+  include Sidekiq::Worker
+  include Sidetiq::Schedulable
+  include MiqToolsServices::SidekiqWorkerMixin
+  sidekiq_options :queue => :miq_bot, :retry => false
+
+  recurrence { hourly.minute_of_hour(0, 5, 10, 15, 20, 25, 30, 35, 40, 45, 50, 55) }
+
+  def perform
+    if !first_unique_worker?
+      logger.info "#{self.class} is already running, skipping"
+    else
+      kill_builds
+    end
+  end
+
+  private
+
+  def kill_builds
+    repo = CommitMonitorRepo.where(:upstream_user => "ManageIQ", :name => "manageiq").first
+    repo.with_travis_service do |travis|
+      builds_to_cancel = travis.builds
+        .take_while { |b| b.pending? || b.canceled? }
+        .reject(&:canceled?)
```

Comment:
Ok, so when we walk the builds it goes by build number order, If we stop at a canceled build, we'll miss a ton of pending builds.

![screen shot 2015-02-25 at 5 44 19 pm](https://cloud.githubusercontent.com/assets/52120/6382373/0a464308-bd16-11e4-9de6-77e3b4c1ce3f.png)

So in that screenshot, if we stop at a canceled build, we would only get the first 8 instead of everything up to the first green one.  Once we have everything, we throw away all the canceled and should be left with pending builds.